src/Speck.js

```diff
@@ -39,12 +39,23 @@ class Speck {
     this._validate();
   }
 
+  __initFieldvalue(field, data) {
+    const hasValue = !isNil(data[field]);
+
+    if (hasValue) return data[field];
+
+    const isFunction = this.schema[field].defaultValue instanceof Function;
+
+    if (isFunction) return this.schema[field].defaultValue.bind(this);
```

Comment:
Is there a better way to "clone" a function? 

I worry because there's no way we can know if our user would want to bind the function this way.